src/background.js

```diff
@@ -1,4 +1,51 @@
 import './activity-logger/background'
+import { filterVisitsByQuery } from './search'
+import niceTime from './util/nice-time'
+
+const makeSuggestion = (query, suggest) => {
+    filterVisitsByQuery({query, limit: 5}).then(searchResult => {
+        const suggestions = []
+        if(searchResult.rows.length === 0) {
+            chrome.omnibox.setDefaultSuggestion({description: 'No results found'})
+        } else {
+            chrome.omnibox.setDefaultSuggestion({description: 'Select an option below'})
+        }
+        searchResult.rows.map((result) => {
+            let doc = result.doc
+            let content = doc.url
+            let title = ''
+            let visitDate = niceTime(doc.visitStart)
+            let url = "<url>" + (content.slice(0,30)) + "</url>"
+            if(url.length > 30) {
+                url+= '...'
+            }
+            if(doc.page.extractedMetadata && doc.page.extractedMetadata.title) {
+                title = "<match>" + escape(doc.page.extractedMetadata.title) + "</match>"
+                title = title.replace(/%20|%2C|%3A|%3F|%B7|%23|%26/gi, matched => ' ')
```

Comment:
escape replaces -, #, spaces with %20, %2C etc so it replaces them with spaces otherwise It'll look ugly.